systems/plants/RigidBodyManipulator.cpp

```diff
@@ -1709,6 +1894,216 @@ void RigidBodyManipulator::forwarddJac(const int body_or_frame_id, const MatrixB
   dJ = dJ_t.transpose();
 }
 
+template <typename DerivedPoints>
+GradientVar<typename DerivedPoints::Scalar, Eigen::Dynamic, DerivedPoints::ColsAtCompileTime> RigidBodyManipulator::forwardKinNew(const MatrixBase<DerivedPoints>& points, int current_body_or_frame_ind, int new_body_or_frame_ind, int rotation_type, int gradient_order)
+{
```

Comment:
An interesting thing here is that if the kinsol has gradients, I can compute J = dx/dq in two ways: 1) using the dTdq from the kinsol, or 2) using forwardKinV (Jacobian output), which does not require kinsol gradients but instead uses the motion subspaces, transformed to world frame. Need to figure out which way is faster if gradients happen to be available.